src/commands/runs/ls.ts

```diff
@@ -0,0 +1,283 @@
+	};
+}
+
+function prettyPrintStatus(status: string) {
+	switch (status) {
+		case 'READY':
+			return chalk.green('Ready');
+		case 'RUNNING':
+			return chalk.blue('Running');
+		case 'SUCCEEDED':
+			return chalk.green('Succeeded');
+		case 'FAILED':
+			return chalk.red('Failed');
+		case 'ABORTING':
+			return chalk.yellow('Aborting');
+		case 'ABORTED':
+			return chalk.red('Aborted');
+		case 'TIMING-OUT':
+			return chalk.yellow('Timing Out');
+		case 'TIMED-OUT':
+			return chalk.red('Timed Out');
+		default:
+			return chalk.gray(
+				(status as string)
+					.split('-')
+					.map((part) => part[0].toUpperCase() + part.slice(1).toLowerCase())
+					.join(' '),
+			);
+	}
+}
+
+const ShortDurationFormatter = new SapphireDurationFormatter({
+	[TimeTypes.Day]: {
+		DEFAULT: 'd',
+	},
+	[TimeTypes.Hour]: {
+		DEFAULT: 'h',
+	},
+	[TimeTypes.Minute]: {
+		DEFAULT: 'm',
+	},
+	[TimeTypes.Month]: {
+		DEFAULT: 'M',
+	},
+	[TimeTypes.Second]: {
+		DEFAULT: 's',
+	},
+	[TimeTypes.Week]: {
+		DEFAULT: 'w',
+	},
+	[TimeTypes.Year]: {
+		DEFAULT: 'y',
+	},
+});
+
+// END of TODO
+
+const multilineTimestampFormatter = new Timestamp(`YYYY-MM-DD[\n]HH:mm:ss`);
+const tableFactory = (compact = false) => {
+	const options: Record<string, unknown> = {
+		head: ['ID', 'Status', 'Results', 'Usage', 'Start Date', 'Took', 'Build Number', 'Origin'],
+		style: {
+			head: ['cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan'],
+			compact,
+		},
+		colAligns: ['middle', 'middle', 'middle', 'middle', 'middle', 'middle', 'middle', 'middle'],
+	};
+
+	if (compact) {
+		options.chars = {
+			'mid': '',
+			'left-mid': '',
+			'mid-mid': '',
+			'right-mid': '',
+			middle: ' ',
+			'top-mid': '─',
+			'bottom-mid': '─',
+		};
+	}
+
+	return new Table<[string, string, string, string, string, string, string, string]>(options);
+};
+
+export class RunsLsCommand extends ApifyCommand<typeof RunsLsCommand> {
+	static override description = 'Lists all runs of the Actor.';
+
+	static override flags = {
+		actor: Flags.string({
+			description:
+				'Optional Actor ID or Name to list runs for. By default, it will use the Actor from the current directory.',
+		}),
+		offset: Flags.integer({
+			description: 'Number of runs that will be skipped.',
+			default: 0,
+		}),
+		limit: Flags.integer({
+			description: 'Number of runs that will be listed.',
+			default: 10,
+		}),
+		desc: Flags.boolean({
+			description: 'Sort runs in descending order.',
+			default: false,
+		}),
+		compact: Flags.boolean({
+			description: 'Display a compact table.',
+			default: false,
+			char: 'c',
+		}),
+	};
+
+	static override enableJsonFlag = true;
+
+	async run() {
+		const { actor, desc, limit, offset, compact, json } = this.flags;
+
+		const client = await getLoggedClientOrThrow();
+
+		// TODO: technically speaking, we don't *need* an actor id to list builds. But it makes more sense to have a table of builds for a specific actor.
```

Comment:
what does this TODO mean? maybe it should be just a regular comment?

Reply:
This has been copied 1:1 from builds (where the same is applicable). We don't _need_ an actor id passed/present to list builds/runs, as we can fetch global list, but it makes more sense to fetch per-actor. Probably best to decide for runs if that still is the case

Reply:
i am actually using [this page](https://console.apify.com/actors/runs) quite often, i dont think we should disallow listing all runs of a given user. i agree with builds, that feels weird without the actor context.

---

what i mean with my comment is that the TODO is not telling me what is missing, what should be done, what is to do, heh